subprojects/configuration-cache/build.gradle.kts

```diff
@@ -16,9 +16,20 @@ dependencies {
 }
 
 tasks.processResources {
-    from(zipTree(provider { configurationCacheReportPath.files.first() })) {
-        into("org/gradle/configurationcache/problems")
-        exclude("META-INF/**")
+    // You can have a faster feedback loop by running `configuration-cache-report` as an included build
+    // See https://github.com/gradle/configuration-cache-report#development-with-gradlegradle-and-composite-build
+    val includedBuild = gradle.includedBuilds.firstOrNull { it.name == "configuration-cache-report" }
```

Comment:
This should not be necessary. Dependency substitution should replace the JAR on the configuration. Does this not work out of the box when you run with --include-build?